gbmgeometry/gbm_detector.py

```diff
@@ -58,20 +60,77 @@ def update_position(self, quaternion, sc_pos=None, time=None):
                                                sc_pos_Y=scy,
                                                sc_pos_Z=scz,
                                                ))
-
+        self._x = SkyCoord(lon=0 * u.deg,
+                           lat=0 * u.deg,
+                           unit='deg',
+                           frame=GBMFrame(quaternion_1=q1,
+                                          quaternion_2=q2,
+                                          quaternion_3=q3,
+                                          quaternion_4=q4,
+                                          sc_pos_X=scx,
+                                          sc_pos_Y=scy,
+                                          sc_pos_Z=scz,
+                                               ))
+        self._y = SkyCoord(lon=90 * u.deg,
+                           lat=0 * u.deg,
+                           unit='deg',
+                           frame=GBMFrame(quaternion_1=q1,
+                                          quaternion_2=q2,
+                                          quaternion_3=q3,
+                                          quaternion_4=q4,
+                                          sc_pos_X=scx,
+                                          sc_pos_Y=scy,
+                                          sc_pos_Z=scz,
+                                          ))
+        self._z = SkyCoord(lon=0 * u.deg,
+                           lat=90 * u.deg,
+                           unit='deg',
+                           frame=GBMFrame(quaternion_1=q1,
+                                          quaternion_2=q2,
+                                          quaternion_3=q3,
+                                          quaternion_4=q4,
+                                          sc_pos_X=scx,
+                                          sc_pos_Y=scy,
+                                          sc_pos_Z=scz,
+                                          ))
         if self._time is not None:
             # we can calculate the sun position
             tmp_sun = get_sun(self._time).icrs
 
-                        
             self._sun_position = SkyCoord(tmp_sun.ra.deg,tmp_sun.dec.deg,unit='deg', frame='icrs').transform_to(self._center.frame)
 
-            tmp_earth = get_body('earth',time=self._time).icrs
-            
-            self._earth_position = SkyCoord(tmp_earth.ra.deg, tmp_earth.dec.deg, unit='deg', frame='icrs').transform_to(self._center.frame)
+            #sun_pos=np.array([np.cos(tmp_sun.ra.deg*(np.pi/180)) * np.cos(tmp_sun.dec.deg*(np.pi/180)),
+            #                          np.sin(tmp_sun.ra.deg*(np.pi/180)) * np.cos(tmp_sun.dec.deg*(np.pi/180)),
+            #                          np.sin(tmp_sun.dec.deg*(np.pi/180))])
+            #self._sun_pos_norm=self.geo_to_gbm(sun_pos/ np.linalg.norm(sun_pos))
+
+            #position of earth in satellite frame:
+
```

Comment:
The earth position norm should probably be private via an underscore. Access within the class can use the underscore. If you want access outside the class, create a property. This way, a user cannot change the value which would screw up future calculations.